pkg/controller/controller_utils.go

```diff
@@ -683,3 +683,13 @@ func BuildClusterAPIMachineSet(ms *clusteroperator.ClusterMachineSet, clusterDep
 
 	return &capiMachineSet, nil
 }
+
+// StringPtrsEqual safely returns true if the value for each string pointer is equal, or both are nil.
+func StringPtrsEqual(s1, s2 *string) bool {
+	if s1 == nil && s2 == nil {
```

Comment:
Getting *really* nit-picky here, but this alternative will save a string comparison if the pointers are equal. Subjectively, the indentation of the flow is better, too, since the expected behavior of comparing the pointed-to values is not indented.

```
if s1 == s2 {
  return true
}
if s1 == nil || s2 == nil {
  return false
}
return *s1 == *s2
```

Reply:
Sounds good, but I think I need to add a check if both are nil, because I need that to return true, before checking if just one is nil and returning false.

Reply:
Nope I don't, passes as is. Thanks!